### ISSUE TITLE
Custom number of steps per epoch in PyTorch

### ISSUE THREAD

**Issue by jbohnslav** [NVIDIA/DALI#1794]
Hi there,

I'm working on a video project using Kinetics700. I use this dataset for pretraining my models, and I don't have enormous amounts of computational resources. I do not have the resources to go through one full epoch of the data before I evaluate (e.g. to checkpoint the model and change learning rate). After building with a reasonable batch size, `pipe.epoch_size()` = more than 12 million. Which is far too many to iterate through. 

I want to use `nvidia.dali.plugin.pytorch.DALIGenericIterator`. I could do something like: 
```python
train_loader = DALIGenericIterator(...)
val_loader = DALIGenericIterator(...)
for epoch in range(epochs):
  for i in range(1000):
      videos, labels = next(train_loader)
      # do stuff

  for i in range(1000):
      videos, labels = next(val_loader)
      # evaluate

  train_loader.reset()
  val_loader.reset()
```
However, the DALIGenericIterator.reset() says: "Resets the iterator after the full epoch. DALI iterators do not support resetting before the end of the epoch and will ignore such request."

Is there any way to use DALI iterators without actually going through your whole training set at once?

**Comment by JanuszL** [comment]
Hi,
Yes, just don't call `.reset` before you go through the whole data set. Also, you can set `auto_reset` in the iterator to `True` and no longer worry when to call `.reset`

**Comment by jbohnslav** [comment]
Hi, thanks very much. That was a simpler solution than I was expecting!
